### PR TITLE
Add troubleshooting section for labeling nodes with reserved labels

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Documentation added/planned (if applicable):*
Add troubleshooting section for labeling nodes with reserved labels

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 1]
```diff
@@ -460,6 +460,30 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
 
+### Labeling nodes with reserved labels such as node-role.kubernetes.io fails with kubeadm error during bootstrap
+
+If cluster creation or upgrade fails to complete successfully and kubelet throws an error similar to the following, please refer this section.
+
+```
+unknown 'kubernetes.io' or 'k8s.io' labels specified with --node-labels: [node-role.kubernetes.io/worker].
+--node-labels in the 'kubernetes.io' namespace must begin with an allowed prefix (kubelet.kubernetes.io, node.kubernetes.io) or be in the specifically allowed set (beta.kubernetes.io/arch, beta.kubernetes.io/instance-type, beta.kubernetes.io/os, failure-domain.beta.kubernetes.io/region, failure-domain.beta.kubernetes.io/zone, kubernetes.io/arch, kubernetes.io/hostname, kubernetes.io/os, node.kubernetes.io/instance-type, topology.kubernetes.io/region, topology.kubernetes.io/zone)
+```
+Self-assigning Node labels such as node-role.kubernetes.io using the kubelet --node-labels flag is not possible due to a security measure imposed by the NodeRestriction admission controller that kubeadm enables by default.
```
Comment (jiayiwang7) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:471]:
```suggestion
Self-assigning Node labels such as `node-role.kubernetes.io` using the kubelet `--node-labels` flag is not possible due to a security measure imposed by the NodeRestriction admission controller that kubeadm enables by default.
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 2]
```diff
@@ -460,6 +460,30 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
+
+Assigning such labels to Nodes can be done after the bootstrap process has completed:
```
Comment (jiayiwang7) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:473]:
```suggestion
Assigning such labels to nodes can be done after the bootstrap process has completed:
```

**Review by jiayiwang7 — COMMENTED** [review]
Can it be more EKS-A specific? I understand you paste the paragraph from CAPI but in our case, as an user I'd like to know for example, what exact EKS-A config can cause this failure (specifying labels in cluster spec), etc.

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 3]
```diff
@@ -460,6 +460,48 @@ To fix it, make sure to validate the network/firewall settings from the workload
 * [Prerequisite Checklist for EKS Anywhere on Snow]({{< relref "../getting-started/snow/snow-getstarted/#prerequisite-checklist" >}}) 
 * [Requirements for EKS Anywhere on Nutanix Cloud Infrastructure]({{< relref "../getting-started/nutanix/nutanix-prereq" >}}) 
 
+### Labeling nodes with reserved labels such as `node-role.kubernetes.io` fails with kubeadm error during bootstrap
+
+If cluster creation or upgrade fails to complete successfully and kubelet throws an error similar to the one below, please refer this section. The cluster spec for the EKSA create or upgrade might look something like:
```
Comment (vivek-koppuru) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:465]:
```suggestion
If cluster creation or upgrade fails to complete successfully and kubelet throws an error similar to the one below, please refer this section. The cluster spec for EKS Anywhere create or upgrade might look something like:
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 4]
```diff
@@ -460,6 +460,48 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
+```
+kubectl label nodes <name> node-role.kubernetes.io/worker=""
+```
+For convenience, here is an example one-liner to do this post installation
+
+```
+# Kubernetes 1.19 (kubeadm 1.19 sets only the node-role.kubernetes.io/master label)
+kubectl get nodes --no-headers -l '!node-role.kubernetes.io/master' -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}' | xargs -I{} kubectl label node {} node-role.kubernetes.io/worker=''
+# Kubernetes >= 1.20 (kubeadm >= 1.20 sets the node-role.kubernetes.io/control-plane label)
+kubectl get nodes --no-headers -l '!node-role.kubernetes.io/control-plane' -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}' | xargs -I{} kubectl label node {} node-role.kubernetes.io/worker=''
```
Comment (vivek-koppuru) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:502]:
```suggestion
# For Kubernetes 1.19 (kubeadm 1.19 sets only the node-role.kubernetes.io/master label)
kubectl get nodes --no-headers -l '!node-role.kubernetes.io/master' -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}' | xargs -I{} kubectl label node {} node-role.kubernetes.io/worker=''

# For Kubernetes >= 1.20 (kubeadm >= 1.20 sets the node-role.kubernetes.io/control-plane label)
kubectl get nodes --no-headers -l '!node-role.kubernetes.io/control-plane' -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}' | xargs -I{} kubectl label node {} node-role.kubernetes.io/worker=''
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 5]
```diff
@@ -460,6 +460,48 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
+--node-labels in the 'kubernetes.io' namespace must begin with an allowed prefix (kubelet.kubernetes.io, node.kubernetes.io) or be in the specifically allowed set (beta.kubernetes.io/arch, beta.kubernetes.io/instance-type, beta.kubernetes.io/os, failure-domain.beta.kubernetes.io/region, failure-domain.beta.kubernetes.io/zone, kubernetes.io/arch, kubernetes.io/hostname, kubernetes.io/os, node.kubernetes.io/instance-type, topology.kubernetes.io/region, topology.kubernetes.io/zone)
+```
+Self-assigning node labels such as `node-role.kubernetes.io` using the kubelet `--node-labels` flag is not possible due to a security measure imposed by the NodeRestriction admission controller that kubeadm enables by default.
+
+Assigning such labels to nodes can be done after the bootstrap process has completed:
+
+```
+kubectl label nodes <name> node-role.kubernetes.io/worker=""
+```
+For convenience, here is an example one-liner to do this post installation
```
Comment (chrisnegus) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:496]:
```suggestion
For convenience, here are example one-liners to do this post installation:
```

**Review by chrisnegus — COMMENTED** [review]
One small suggestion.

**Review by drewvanstone — APPROVED** [review]

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 6]
```diff
@@ -460,6 +460,49 @@ To fix it, make sure to validate the network/firewall settings from the workload
 * [Prerequisite Checklist for EKS Anywhere on Snow]({{< relref "../getting-started/snow/snow-getstarted/#prerequisite-checklist" >}}) 
 * [Requirements for EKS Anywhere on Nutanix Cloud Infrastructure]({{< relref "../getting-started/nutanix/nutanix-prereq" >}}) 
 
+### Labeling nodes with reserved labels such as `node-role.kubernetes.io` fails with kubeadm error during bootstrap
+
+If cluster creation or upgrade fails to complete successfully and kubelet throws an error similar to the one below, please refer this section. The cluster spec for EKS Anywhere create or upgrade might look something like:
```
Comment (drewvanstone) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:465]:
```suggestion
If cluster creation or upgrade fails to complete successfully and kubelet throws an error similar to the one below, please refer to this section. The cluster spec for EKS Anywhere create or upgrade should look like:
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 7]
```diff
@@ -460,6 +460,49 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
+        "node-role.kubernetes.io/control-plane": "cp"
+   workerNodeGroupConfigurations: 
+   - count: 2 
+     labels:                        
+        "node-role.kubernetes.io/worker": "worker"
+.
+.
+```
+
+If your cluster spec looks like the above one for either the control plane configuration and/or worker node configuration, you might run into the below kubelet error.
```
Comment (drewvanstone) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:484]:
```suggestion
If your cluster spec looks like the above one for either the control plane configuration and/or worker node configuration, you might run into the below kubelet error:
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 8]
```diff
@@ -460,6 +460,49 @@ To fix it, make sure to validate the network/firewall settings from the workload
...
+--node-labels in the 'kubernetes.io' namespace must begin with an allowed prefix (kubelet.kubernetes.io, node.kubernetes.io) or be in the specifically allowed set (beta.kubernetes.io/arch, beta.kubernetes.io/instance-type, beta.kubernetes.io/os, failure-domain.beta.kubernetes.io/region, failure-domain.beta.kubernetes.io/zone, kubernetes.io/arch, kubernetes.io/hostname, kubernetes.io/os, node.kubernetes.io/instance-type, topology.kubernetes.io/region, topology.kubernetes.io/zone)
+```
+Self-assigning node labels such as `node-role.kubernetes.io` using the kubelet `--node-labels` flag is not possible due to a security measure imposed by the NodeRestriction admission controller that kubeadm enables by default.
+
+Assigning such labels to nodes can be done after the bootstrap process has completed:
+
+```
+kubectl label nodes <name> node-role.kubernetes.io/worker=""
+```
+For convenience, here are example one-liners to do this post installation:
```
Comment (drewvanstone) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:496]:
```suggestion
For convenience, here are example one-liners to do this post-installation:
```

**Review by drewvanstone — APPROVED** [review]

**Pull request merged**